Guide-on-defining-types.md

```diff
@@ -749,3 +749,39 @@ fingerprintHashesIndex.forEach((hashIndex: number) => {
       fingerprint.push([kGramHashes[hashIndex], hashIndex]);
 });
 ```
+
+### [TS-9]:
+ The operand of a 'delete' operator must be optional.
+#### [TS-9-1]:
+##### Violation:
+This error occurs when using the delete operator in strictNullChecks. Because the property is no longer implemented after deletion, set its type to undefined. As a result, the property must be undefined; otherwise, the delete operator leads to an error.
```

Comment:
```suggestion
This error occurs when using the `delete` operator with strictNullChecks. Since a property cannot be implemented after deletion, its type is set to undefined. So, the property must contain undefined as a type, otherwise the delete operator leads to an error.
```

Reply:
Done